src/libraries/Fuzzing/DotnetFuzzing/Fuzzers/Base64Fuzzer.cs

```diff
@@ -0,0 +1,105 @@
+
+        public void FuzzTarget(ReadOnlySpan<byte> bytes)
+        {
+            using PooledBoundedMemory<byte> inputPoisoned = PooledBoundedMemory<byte>.Rent(bytes, PoisonPagePlacement.After);
+            Span<byte> input = inputPoisoned.Span;
+            int maxEncodedLength = Base64.GetMaxEncodedToUtf8Length(bytes.Length);
+            using PooledBoundedMemory<byte> destPoisoned = PooledBoundedMemory<byte>.Rent(maxEncodedLength, PoisonPagePlacement.After);
+            Span<byte> encoderDest = destPoisoned.Span;
+
+            { // IsFinalBlock = true
+                OperationStatus status = Base64.EncodeToUtf8(input, encoderDest, out int bytesConsumed, out int bytesEncoded);
+
+                Assert.Equal(OperationStatus.Done, status);
+                Assert.Equal(bytes.Length, bytesConsumed);
+                Assert.Equal(true, maxEncodedLength >= bytesEncoded);       
+
+                using PooledBoundedMemory<byte> decoderDestPoisoned = PooledBoundedMemory<byte>.Rent(Base64.GetMaxDecodedFromUtf8Length(bytesEncoded), PoisonPagePlacement.After);
+                status = Base64.DecodeFromUtf8(encoderDest.Slice(0, bytesEncoded), decoderDestPoisoned.Span, out int bytesRead, out int bytesDecoded);
+
+                Assert.Equal(OperationStatus.Done, status);
+                Assert.Equal(bytes.Length, bytesDecoded);
+                Assert.Equal(bytesEncoded, bytesRead);
+                Assert.SequenceEqual(bytes, decoderDestPoisoned.Span.Slice(0, bytesDecoded));
+            }
+
+            { // IsFinalBlock = false
+                encoderDest.Clear();
+                OperationStatus status = Base64.EncodeToUtf8(input, encoderDest, out int bytesConsumed, out int bytesEncoded, isFinalBlock: false);
+                Span<byte> decodeInput = encoderDest.Slice(0, bytesEncoded);
+                using PooledBoundedMemory<byte> decoderDestPoisoned = PooledBoundedMemory<byte>.Rent(Base64.GetMaxDecodedFromUtf8Length(bytesEncoded), PoisonPagePlacement.After);
+                Span<byte> decoderDest = decoderDestPoisoned.Span;
+
+                if (bytes.Length % 3 == 0)
+                {
+                    Assert.Equal(OperationStatus.Done, status);
+                    Assert.Equal(bytes.Length, bytesConsumed);
+                    Assert.Equal(true, maxEncodedLength >= bytesEncoded);
+
+
+                    status = Base64.DecodeFromUtf8(decodeInput, decoderDest, out int bytesRead, out int bytesDecoded, isFinalBlock: false);
+
+                    Assert.Equal(OperationStatus.Done, status);
+                    Assert.Equal(bytes.Length, bytesDecoded);
+                    Assert.Equal(bytesEncoded, bytesRead);
+                    Assert.SequenceEqual(bytes, decoderDest.Slice(0, bytesDecoded));
+                }
+                else
+                {
+                    Assert.Equal(OperationStatus.NeedMoreData, status);
+                    Assert.Equal(true, maxEncodedLength >= bytesEncoded);
```

Comment:
Should this assert the exact expected value instead?

E.g. `Assert.Equal(input.Length / 3 * 4, bytesEncoded)` (or whichever formula applies here)

Reply:
The formula does not return exact encoded value, exact encoded value can be equal to `maxEncodedLength`, `maxEncodedLength  - 1`, or `maxEncodedLength - 2` depending on padding count https://github.com/dotnet/runtime/blob/db5e98d64641784213f5e88b3d1523c7ef47c01d/src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Base64Encoder.cs#L174

I can update the conditions with ORs

Reply:
I would expect that with `OperationStatus.NeedMoreData`, the remainder of the input shouldn't be consumed.
After an `EncodeToUtf8` call, `bytesConsumed` should be divisible by 3, meaning that `written` would be `consumed / 3 * 4`.
Is that not the case?

Reply:
Ah I see you meant for this exact case not in general, yes, you are right